biosys/apps/main/tests/api/test_filtering_and_ordering.py

```diff
@@ -0,0 +1,622 @@
+            ['Species Name', 'When', 'Latitude', 'Longitude', 'Where'],
+            ['Canis lupus', '2018-06-22', -24.882248, 113.662995, 'Carnavon'],
+            ['Quokka', '2018-04-01', -23.142194, 113.772222, 'Coral Bay'],
+            ['Eucalyptus robusta', '2017-11-23', -21.177125, 119.765361, 'Marble Bar'],
+            ['Chubby Bat', '2018-08-23', -17.962075, 122.234554, 'Broome']
+        ])
+
+    def test_get_record_happy(self):
+        """
+        Test GET /records/ endpoint
+        """
+        url = reverse('api:record-list')
+        client = self.custodian_1_client
+
+        # query with polygon of Perth roughly. That should exclude mandurah and all the north records
+        west = 115.0
+        east = 116.0
+        north = -31.5
+        south = -32.1  # north of Mandurah
+
+        query = {
+            'geometry__within': Polygon.from_bbox((west, south, east, north)).geojson
+        }
+        expected_records = [
+            ['Canis lupus', '2018-06-22', -32, 115.75, 'Cottesloe'],
+            ['Quokka', '2018-04-01', -32.011935, 115.517554, 'Rottnest Island'],
+            ['Eucalyptus robusta', '2017-11-23', -31.959765, 115.8322754, 'Kings Park'],
+        ]
+        response = client.get(url, query)
+        self.assertEqual(200, response.status_code)
+        records = response.json()
+        self.assertIsInstance(records, list)
+        self.assertEqual(len(records), len(expected_records))
+        # check location
+        self.assertEqual(
+            sorted([r['data'].get('Where') for r in records]),
+            ['Cottesloe', 'Kings Park', 'Rottnest Island']
+        )
+
+    def test_get_dataset_records_happy(self):
+        """
+        Same test as above but GET /dataset/{id}/records/
+        """
+        url = reverse('api:dataset-records', kwargs={'pk': self.perth_metro_ds.pk})
+        client = self.custodian_1_client
+
+        # query with polygon of Perth roughly. That should exclude mandurah and all the north records
+        west = 115.0
+        east = 116.0
+        north = -31.5
+        south = -32.1  # north of Mandurah
+
+        query = {
+            'geometry__within': Polygon.from_bbox((west, south, east, north)).geojson
+        }
+        expected_records = [
+            ['Canis lupus', '2018-06-22', -32, 115.75, 'Cottesloe'],
+            ['Quokka', '2018-04-01', -32.011935, 115.517554, 'Rottnest Island'],
+            ['Eucalyptus robusta', '2017-11-23', -31.959765, 115.8322754, 'Kings Park'],
+        ]
+        response = client.get(url, query)
+        self.assertEqual(200, response.status_code)
+        records = response.json()
+        self.assertIsInstance(records, list)
+        self.assertEqual(len(records), len(expected_records))
+        # check location
+        self.assertEqual(
+            sorted([r['data'].get('Where') for r in records]),
+            ['Cottesloe', 'Kings Park', 'Rottnest Island']
+        )
```

Comment:
I think this test should have a bounding box that includes both datasets, so as to ensure you only get the records from the dataset you specify.

Reply:
Yep good idea.